### ISSUE TITLE
TypeError: this._parser.startDataMode is not a function

### ISSUE THREAD

**Issue by Fusl** [andris9/smtp-server#50]
I've recently noticed my app crashing every now and then with the error message below.

```
/opt/eftrap/node_modules/smtp-server/lib/smtp-connection.js:945
    this._dataStream = this._parser.startDataMode(this._server.options.size);
                                    ^

TypeError: this._parser.startDataMode is not a function
    at SMTPConnection.handler_DATA (/opt/eftrap/node_modules/smtp-server/lib/smtp-connection.js:945:37)
    at SMTPConnection._onCommand (/opt/eftrap/node_modules/smtp-server/lib/smtp-connection.js:354:13)
    at SMTPStream.<anonymous> (/opt/eftrap/node_modules/smtp-server/lib/smtp-stream.js:134:18)
    at SMTPConnection.<anonymous> (/opt/eftrap/node_modules/smtp-server/lib/smtp-connection.js:932:9)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

My guess is that we're running into some kind of race condition when a client disconnects unexpectedly at some point.

**Comment by Fusl** [comment]
Another crash, this time with logger set to true:

```
[2016-04-25 23:08:44] INFO: [/HZ9k44t0HDD] Connection from [::ffff:x.x.x.x]
[2016-04-25 23:08:44] DEBUG: [/HZ9k44t0HDD] S: 220 smtp.addr.es ESMTP
[2016-04-25 23:08:44] DEBUG: [qpRENVgrSEsX] C: MAIL From:<xxx@xxx.xxx>
[2016-04-25 23:08:44] DEBUG: [qpRENVgrSEsX] C: RCPT To:<xxx@xxx.xxx>
[2016-04-25 23:08:44] INFO: [qpRENVgrSEsX] Connection closed to [::ffff:x.x.x.x]
[2016-04-25 23:08:44] DEBUG: [qpRENVgrSEsX] C: DATA
/opt/eftrap/node_modules/smtp-server/lib/smtp-connection.js:945
    this._dataStream = this._parser.startDataMode(this._server.options.size);
                                    ^

TypeError: this._parser.startDataMode is not a function
    at SMTPConnection.handler_DATA (/opt/eftrap/node_modules/smtp-server/lib/smtp-connection.js:945:37)
    at SMTPConnection._onCommand (/opt/eftrap/node_modules/smtp-server/lib/smtp-connection.js:354:13)
    at SMTPStream.<anonymous> (/opt/eftrap/node_modules/smtp-server/lib/smtp-stream.js:134:18)
    at SMTPConnection.<anonymous> (/opt/eftrap/node_modules/smtp-server/lib/smtp-connection.js:932:9)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

It seems, we're running into a race condition there when the client sends a DATA string and instantly closes the connection.

**Comment by andris9** [comment]
Yes, it seems that `this._parser` is already set to `false` by the time it is called. I added an extra check for it (https://github.com/andris9/smtp-server/commit/4b50fa75d2150c3ecb8cdcc41de9ca6a9c70987f#diff-9817182324e1860ad3ed5132780936dcR945) and published the code to npm as v1.9.1
